packages/api/engine/openseatypes.go

```diff
@@ -0,0 +1,557 @@
+	PaymentToken       Address   `json:"payment_token"`
+	// PaymentTokenContract PaymentTokenContract `json:"payment_token_contract"`
+	BasePrice       Number `json:"base_price"`
+	Extra           Number `json:"extra"`
+	Quantity        string `json:"quantity"`
+	Salt            Number `json:"salt"`
+	V               *uint8 `json:"v"`
+	R               *Bytes `json:"r"`
+	S               *Bytes `json:"s"`
+	ApprovedOnChain bool   `json:"approved_on_chain"`
+	Cancelled       bool   `json:"cancelled"`
+	Finalized       bool   `json:"finalized"`
+	MarkedInvalid   bool   `json:"marked_invalid"`
+	// PrefixedHash         string               `json:"prefixed_hash"`
+}
+
+// IsPrivate returns true if the order is private
+func (o Order) IsPrivate() bool {
+	if o.Taker.Address != NullAddress {
+		return true
+	}
+	return false
+}
+
+// Side is the direction of an order
+type Side uint8
+
+const (
+	// Buy is a buy order
+	Buy Side = iota
+
+	// Sell is a sell order
+	Sell
+)
+
+// SaleKind is the kind of sale
+type SaleKind uint8
+
+const (
+	// FixedOrMinBid fixed or min bid sale
+	FixedOrMinBid SaleKind = iota // 0 for fixed-price sales or min-bid auctions
+
+	// DutchAuctions is a dutch auction
+	DutchAuctions // 1 for declining-price Dutch Auctions
+)
+
+// HowToCall is the type of call
+type HowToCall uint8
+
+const (
+	// Call is a regular call
+	Call HowToCall = iota
+
+	// DelegateCall is a delegate call
+	DelegateCall
+)
+
+// FeeMethod is the fee method
+type FeeMethod uint8
+
+const (
+	// ProtocolFee is the protocol fee method
+	ProtocolFee FeeMethod = iota
+
+	// SplitFee is the split fee method
+	SplitFee
+)
+
+// type Metadata struct {
+// 	Asset  MetadataAsset `json:"asset"`
+// 	Schema string        `json:"schema"`
+// }
+
+// type MetadataAsset struct {
+// 	ID      string `json:"id"`
+// 	Address string `json:"address"`
+// }
+
+// GetOrders fetches the orders
+func (o Opensea) GetOrders(assetContractAddress string, listedAfter int64) ([]*Order, error) {
+	ctx := context.TODO()
+	return o.GetOrdersWithContext(ctx, assetContractAddress, listedAfter)
+}
+
+// GetOrdersWithContext fetches the orders with context
+func (o Opensea) GetOrdersWithContext(ctx context.Context, assetContractAddress string, listedAfter int64) (orders []*Order, err error) {
+	offset := 0
+	limit := 100
+
+	q := url.Values{}
+	q.Set("asset_contract_address", assetContractAddress)
+	q.Set("listed_after", fmt.Sprintf("%d", listedAfter))
+	q.Set("limit", fmt.Sprintf("%d", limit))
+	q.Set("order_by", "created_date")
+	q.Set("order_direction", "asc")
+
+	orders = []*Order{}
+
+	for true {
+		q.Set("offset", fmt.Sprintf("%d", offset))
+		path := "/wyvern/v1/orders?" + q.Encode()
+		b, err := o.getPath(ctx, path)
+		if err != nil {
+			return nil, err
+		}
+
+		out := &struct {
+			Count  int64    `json:"count"`
+			Orders []*Order `json:"orders"`
+		}{}
+
+		err = json.Unmarshal(b, out)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Limit err scope when possible
```suggestion
		if err := json.Unmarshal(b, out); err != nil {
			return nil, err
		}
```